### PR TITLE
chore(test):  introduce a parent class that encapsulates the shared functionality of all cluster creation pages

### LINKED ISSUE #10372: Add a parent class that encapsulates the shared functionality of all cluster creation pages
### Is your enhancement related to a problem? Please describe

Currently, we have a cluster creation page class only for the "kind" extension. However, there are at least two other extensions that also have cluster creation pages with similar functionality (minikube, OpenShift Local).

### Describe the solution you'd like

It would be beneficial to introduce a parent class that encapsulates the shared functionality of all cluster creation pages across extensions. This would help reduce code duplication and streamline the process of implementing tests for other extensions.

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by amisskii**
Signed-off-by: Anton Misskii <amisskii@redhat.com>

### What does this PR do?

 introduces a parent class that encapsulates the shared functionality of all cluster creation pages

### What issues does this PR fix or reference?

https://github.com/podman-desktop/podman-desktop/issues/10372

### How to test this PR?

- [ ] Tests are covering the bug fix or the new feature

### CODE REVIEW

File: tests/playwright/src/model/pages/cluster-creation-base-page.ts [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+
+import test, { expect as playExpect, type Locator, type Page } from '@playwright/test';
+
+import { BasePage } from './base-page';
+
+export class CreateClusterBasePage extends BasePage {
```
Comment (benoitf) [inline tests/playwright/src/model/pages/cluster-creation-base-page.ts]: question: should it be flagged as abstract ?

**Review by benoitf — APPROVED** [review]
some nit

File: tests/playwright/src/model/pages/cluster-creation-base-page.ts (hunk 1 above)
Comment (amisskii) [reply tests/playwright/src/model/pages/cluster-creation-base-page.ts]: yes,  it should. thank you :)

**Review by cbr7 — APPROVED** [review]

**Review by odockal — APPROVED** [review]
LGTM.

Follow up: after discussion the changes, we found out multiple places that share the functionality and definition in Podman Machine and Kind/Minikube/OpenShift cluster creation and so the next step would be to refactor this code into logical hierarchy.

**Pull request merged**